### PR TITLE
PUB-65 : ⚰️ remove dead and deprecated code

### PR DESCRIPTION
**Pull request by rohit-gohri**
Closes PUB-65

### CODE REVIEW

File: src/service/subscriberLoader.ts [hunk 1]
```diff
@@ -1,15 +1,9 @@
 import { resolve, join } from 'path';
-import fs = require('fs');
-import {
-  SubscriberV2,
-  Subscribers,
-  SubscriberTuple,
-  SubscriberV1,
-} from '../subscriber';
+import fs from 'fs';
+import { SubscriberV2, Subscribers, SubscriberTuple } from '../subscriber';
```
Comment (rohit-gohri) [inline src/service/subscriberLoader.ts:3]: Codacy has a fix for the issue: [Replace `V2,·Subscribers,·SubscriberTuple` with `s,·SubscriberTuple,·SubscriberV2`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
import { Subscribers, SubscriberTuple, SubscriberV2 } from '../subscriber';
```

File: __tests__/pubsub/subscriptions/test-topic.example.subscription.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { SubscriberObject } from '@honestfoodcompany/pubsub';
+
+const TestSubscription: SubscriberObject = {
+  topicName: 'test-topic',
+  subscriptionName: 'test-topic.example.subscription',
+  description: 'Just a test subscription',
+  async handleMessage(message: any): Promise<void> {
```
Comment (rohit-gohri) [inline __tests__/pubsub/subscriptions/test-topic.example.subscription.ts:7]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  async handleMessage(message: unknown): Promise<void> {
```

File: __tests__/subscription.test.ts [hunk 3]
```diff
@@ -1,48 +1,64 @@
...
-describe('subscription', () => {
-  it('getters should return correct values', async () => {
-    const sub = new TestSubscription();
-    expect(TestSubscription.topicName).toBe('test-topic');
-    expect(TestSubscription.subscriptionName).toBe('test-topic.subscription');
-    expect(TestSubscription.description).toBe('Just a test subscription');
-    expect(typeof sub.handleMessage).toBe('function');
-    expect(TestSubscription.ackDeadlineSeconds).toBe(10);
-    expect(TestSubscription.maxMessages).toBe(1);
+describe('@subscription', (): any => {
```
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:25]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
describe('@subscription', (): unknown => {
```

File: __tests__/subscription.test.ts [hunk 4]
```diff
@@ -1,48 +1,64 @@
...
-  PubSub: jest.fn().mockImplementation(() => ({
-    topic: mockTopic,
-    publish: jest.fn(),
-  })),
+  PubSub: mockPubSub,
 }));
 
-jest.mock('../src/service/pubsub', () => ({
+const mockPublish = jest.fn();
+jest.mock('../src/service/pubsub', (): any => ({
```
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:12]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
jest.mock('../src/service/pubsub', (): unknown => ({
```

File: __tests__/subscription.test.ts [hunk 5]
```diff
@@ -1,48 +1,64 @@
...
+  it('should allow the default values to be overridden', () => {
+    const subscription = subscriptions.find((sub) => {
+      const [, { subscriptionName }] = sub;
+      return (
+        subscriptionName === 'test-topic.example.override-options.subscription'
+      );
+    });
+
+    const subscriptionObj =
+      subscription && subscription.length > 1 ? subscription[1] : ({} as any);
```
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:60]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
      subscription && subscription.length > 1 ? subscription[1] : ({} as unknown);
```

File: __tests__/subscription.with-custom-credentials.test.ts [hunk 6]
```diff
@@ -1,6 +1,8 @@
-import PubSubService from '../src/service/pubsub';
-import { Subscribers, SubscriberTuple } from '../src/subscriber';
-import { SubscriptionService } from '../src';
+import {
+  PubSubService,
+  SubscriptionService,
```
Comment (rohit-gohri) [inline __tests__/subscription.with-custom-credentials.test.ts:3]: Codacy has a fix for the issue: [Replace `ptionServic` with `berTupl`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  SubscriberTuple,
```

File: src/service/subscriberLoader.ts (hunk 1 above)
Comment (rohit-gohri) [inline src/service/subscriberLoader.ts:3]: Codacy has a fix for the issue: [Replace `V2,·Subscribers,·SubscriberTuple` with `s,·SubscriberTuple,·SubscriberV2`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
import { Subscribers, SubscriberTuple, SubscriberV2 } from '../subscriber';
```

File: __tests__/subscription.test.ts [hunk 7]
```diff
@@ -1,48 +1,64 @@
...
+
+  it('should have default options when not specified', () => {
+    const subscription = subscriptions.find((sub) => {
+      const [, { subscriptionName }] = sub;
+      return subscriptionName === 'test-topic.example.subscription';
+    });
+
+    const subscriptionObj = subscription?.length
+      ? subscription[1]
+      : ({} as any);
```
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:45]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
      : ({} as unknown);
```

File: __tests__/subscriptions.service.test.ts [hunk 8]
```diff
@@ -21,21 +21,24 @@ jest.mock('../src/service/pubsub', (): any => ({
...
-    },
-    init: function (): any {
-      return this;
-    },
-    start: jest.fn(),
-  })),
-}));
+jest.mock(
+  './pubsub/subscriptions/test-topic.example.subscription',
+  (): any => ({
```
Comment (rohit-gohri) [inline __tests__/subscriptions.service.test.ts:26]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  (): unknown => ({
```

File: __tests__/pubsub/subscriptions/test-topic.example.subscription.ts (hunk 2 above)
Comment (rohit-gohri) [inline __tests__/pubsub/subscriptions/test-topic.example.subscription.ts:7]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  async handleMessage(message: unknown): Promise<void> {
```

File: __tests__/subscription.test.ts [hunk 9]
```diff
@@ -1,48 +1,64 @@
...
+jest.mock('../src/service/pubsub', (): any => ({
+  __esModule: true,
   default: class {
-    public static getInstance() {
+    public static getInstance(): any {
+      console.log('getting instance');
       return new this();
     }
-    public publish() {
+    public publish(): any {
```
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:19]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
    public publish(): unknown {
```

File: __tests__/subscription.with-custom-credentials.test.ts [hunk 10]
```diff
@@ -1,6 +1,8 @@
...
+  SubscriberTuple,
```
Comment (rohit-gohri) [inline __tests__/subscription.with-custom-credentials.test.ts:4]: Codacy has a fix for the issue: [Replace `berTupl` with `ptionServic`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  SubscriptionService,
```

File: __tests__/subscription.test.ts [hunk 11]
```diff
@@ -1,48 +1,64 @@
...
-    expect(typeof sub.handleMessage).toBe('function');
-    expect(TestSubscription.ackDeadlineSeconds).toBe(10);
-    expect(TestSubscription.maxMessages).toBe(1);
+describe('@subscription', (): any => {
+  let subscriptions: Subscribers;
+  beforeAll(() => {
+    subscriptions = SubscriptionService.getSubscribers();
+  });
+
+  it('should find subscription', async (): Promise<any> => {
```
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:31]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  it('should find subscription', async (): Promise<unknown> => {
```

File: __tests__/subscription.test.ts [hunk 12]
```diff
@@ -1,48 +1,64 @@
...
+  PubSub: mockPubSub,
 }));
 
-jest.mock('../src/service/pubsub', () => ({
+const mockPublish = jest.fn();
+jest.mock('../src/service/pubsub', (): any => ({
+  __esModule: true,
   default: class {
-    public static getInstance() {
+    public static getInstance(): any {
```
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:15]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
    public static getInstance(): unknown {
```

File: __tests__/subscription.test.ts [hunk 13]
```diff
@@ -1,48 +1,64 @@
...
+  it('should allow the default values to be overridden', () => {
+    const subscription = subscriptions.find((sub) => {
+      const [, { subscriptionName }] = sub;
+      return (
+        subscriptionName === 'test-topic.example.override-options.subscription'
+      );
+    });
+
+    const subscriptionObj =
+      subscription && subscription.length > 1 ? subscription[1] : ({} as any);
```
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:60]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
      subscription && subscription.length > 1 ? subscription[1] : ({} as unknown);
```

File: __tests__/subscription.with-custom-credentials.test.ts (hunk 6 above)
Comment (rohit-gohri) [inline __tests__/subscription.with-custom-credentials.test.ts:3]: Codacy has a fix for the issue: [Replace `ptionServic` with `berTupl`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  SubscriberTuple,
```

File: __tests__/pubsub.service.google.test.ts [hunk 14]
```diff
@@ -18,35 +18,40 @@ const mockAllUsersList = [
...
-    public static getInstance(): any {
+    public static getInstance() {
       return new this();
     }
-    public publish(...args: any[]): any {
+    public publish(...args: any[]) {
       return mockPublish(...args);
     }
-    public subscribe(...args: any[]): any {
+    public subscribe(...args: any[]) {
```
Comment (rohit-gohri) [inline __tests__/pubsub.service.google.test.ts:27]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
    public subscribe(...args: unknown[]) {
```

File: __tests__/subscription.test.ts [hunk 15]
```diff
@@ -1,48 +1,64 @@
...
+
+  it('should have default options when not specified', () => {
+    const subscription = subscriptions.find((sub) => {
+      const [, { subscriptionName }] = sub;
+      return subscriptionName === 'test-topic.example.subscription';
+    });
+
+    const subscriptionObj = subscription?.length
+      ? subscription[1]
+      : ({} as any);
```
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:45]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
      : ({} as unknown);
```

File: __tests__/pubsub.service.google.test.ts [hunk 16]
```diff
@@ -18,35 +18,40 @@ const mockAllUsersList = [
 jest.mock('../src/client/googlePubSub', () => ({
   __esModule: true,
   default: class {
-    public static getInstance(): any {
+    public static getInstance() {
       return new this();
     }
-    public publish(...args: any[]): any {
+    public publish(...args: any[]) {
```
Comment (rohit-gohri) [inline __tests__/pubsub.service.google.test.ts:24]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
    public publish(...args: unknown[]) {
```

File: src/service/subscriberLoader.ts (hunk 1 above)
Comment (rohit-gohri) [inline src/service/subscriberLoader.ts:3]: Codacy has a fix for the issue: [Replace `V2,·Subscribers,·SubscriberTuple` with `s,·SubscriberTuple,·SubscriberV2`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
import { Subscribers, SubscriberTuple, SubscriberV2 } from '../subscriber';
```

File: __tests__/pubsub/subscriptions/test-topic.example.subscription.ts (hunk 2 above)
Comment (rohit-gohri) [inline __tests__/pubsub/subscriptions/test-topic.example.subscription.ts:7]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  async handleMessage(message: unknown): Promise<void> {
```

File: __tests__/subscription.with-custom-credentials.test.ts (hunk 10 above)
Comment (rohit-gohri) [inline __tests__/subscription.with-custom-credentials.test.ts:4]: Codacy has a fix for the issue: [Replace `berTupl` with `ptionServic`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  SubscriptionService,
```

File: __tests__/pubsub/subscriptions/test-topic.example.subscription.ts (hunk 2 above)
Comment (rohit-gohri) [inline __tests__/pubsub/subscriptions/test-topic.example.subscription.ts:7]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  async handleMessage(message: unknown): Promise<void> {
```

File: __tests__/subscription.with-custom-credentials.test.ts (hunk 6 above)
Comment (rohit-gohri) [inline __tests__/subscription.with-custom-credentials.test.ts:3]: Codacy has a fix for the issue: [Replace `ptionServic` with `berTupl`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  SubscriberTuple,
```

File: __tests__/pubsub.service.google.test.ts (hunk 16 above)
Comment (rohit-gohri) [inline __tests__/pubsub.service.google.test.ts:24]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
    public publish(...args: unknown[]) {
```

File: __tests__/subscription.test.ts (hunk 13 above)
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:60]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
      subscription && subscription.length > 1 ? subscription[1] : ({} as unknown);
```

File: __tests__/pubsub.service.google.test.ts (hunk 14 above)
Comment (rohit-gohri) [inline __tests__/pubsub.service.google.test.ts:27]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
    public subscribe(...args: unknown[]) {
```

File: __tests__/subscription.with-custom-credentials.test.ts (hunk 10 above)
Comment (rohit-gohri) [inline __tests__/subscription.with-custom-credentials.test.ts:4]: Codacy has a fix for the issue: [Replace `berTupl` with `ptionServic`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  SubscriptionService,
```

File: src/service/subscriberLoader.ts (hunk 1 above)
Comment (rohit-gohri) [inline src/service/subscriberLoader.ts:3]: Codacy has a fix for the issue: [Replace `V2,·Subscribers,·SubscriberTuple` with `s,·SubscriberTuple,·SubscriberV2`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
import { Subscribers, SubscriberTuple, SubscriberV2 } from '../subscriber';
```

File: __tests__/subscription.test.ts (hunk 15 above)
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:45]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
      : ({} as unknown);
```

File: __tests__/subscription.with-custom-credentials.test.ts (hunk 10 above)
Comment (rohit-gohri) [inline __tests__/subscription.with-custom-credentials.test.ts:4]: Codacy has a fix for the issue: [Replace `berTupl` with `ptionServic`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  SubscriptionService,
```

File: __tests__/pubsub.service.google.test.ts (hunk 14 above)
Comment (rohit-gohri) [inline __tests__/pubsub.service.google.test.ts:27]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
    public subscribe(...args: unknown[]) {
```

File: __tests__/pubsub/subscriptions/test-topic.example.subscription.ts (hunk 2 above)
Comment (rohit-gohri) [inline __tests__/pubsub/subscriptions/test-topic.example.subscription.ts:7]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  async handleMessage(message: unknown): Promise<void> {
```

File: __tests__/pubsub.service.google.test.ts (hunk 16 above)
Comment (rohit-gohri) [inline __tests__/pubsub.service.google.test.ts:24]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
    public publish(...args: unknown[]) {
```

File: __tests__/subscription.with-custom-credentials.test.ts (hunk 6 above)
Comment (rohit-gohri) [inline __tests__/subscription.with-custom-credentials.test.ts:3]: Codacy has a fix for the issue: [Replace `ptionServic` with `berTupl`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
  SubscriberTuple,
```

File: __tests__/subscription.test.ts (hunk 15 above)
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:45]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
      : ({} as unknown);
```

File: __tests__/subscription.test.ts (hunk 13 above)
Comment (rohit-gohri) [inline __tests__/subscription.test.ts:60]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
      subscription && subscription.length > 1 ? subscription[1] : ({} as unknown);
```

File: src/service/subscriberLoader.ts (hunk 1 above)
Comment (rohit-gohri) [inline src/service/subscriberLoader.ts:3]: Codacy has a fix for the issue: [Replace `V2,·Subscribers,·SubscriberTuple` with `s,·SubscriberTuple,·SubscriberV2`](https://app.codacy.com/gh/deliveryhero/hfc-pubsub/pullRequest?prid=8106467)
```suggestion
import { Subscribers, SubscriberTuple, SubscriberV2 } from '../subscriber';
```

File: __tests__/cli.subscriptions.test.ts [hunk 17]
```diff
@@ -46,8 +45,12 @@ function cli(args: any, cwd: string | undefined = undefined): any {
     );
   });
 }
-describe('subscriptions cli', (): any => {
-  it('should find config', async (): Promise<any> => {
+
+jest.setTimeout(10000);
```
Comment (ishan123456789) [inline __tests__/cli.subscriptions.test.ts:49]: Is the delay being used for loading subscriptions through `fs`

**Review by ishan123456789 — APPROVED** [review]
LGTM

File: __tests__/cli.subscriptions.test.ts (hunk 17 above)
Comment (rohit-gohri) [reply __tests__/cli.subscriptions.test.ts:49]: This is for running the CLI through ts-node, it takes some time to transpile src code. Previously dist code was being loaded (dist could be outdated and tests might pass accidentally)

**Pull request merged**